packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -0,0 +1,390 @@
+import type {
+  AccountsControllerAccountAddedEvent,
+  AccountsControllerAccountRemovedEvent,
+  AccountsControllerListMultichainAccountsAction,
+} from '@metamask/accounts-controller';
+import {
+  BaseController,
+  type ControllerGetStateAction,
+  type ControllerStateChangeEvent,
+  type RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { isEvmAccountType } from '@metamask/keyring-api';
+import type { Balance, CaipAssetType } from '@metamask/keyring-api';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import { KeyringClient } from '@metamask/keyring-snap-client';
+import type { HandleSnapRequest } from '@metamask/snaps-controllers';
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+import type { Json, JsonRpcRequest } from '@metamask/utils';
+import type { Draft } from 'immer';
+
+import {
+  BalancesTracker,
+  BALANCE_UPDATE_INTERVALS,
+  NETWORK_ASSETS_MAP,
+} from '.';
+import { getScopeForAccount } from './utils';
+
+const controllerName = 'MultichainBalancesController';
+
+/**
+ * State used by the {@link MultichainBalancesController} to cache account balances.
+ */
+export type MultichainBalancesControllerState = {
+  balances: {
+    [account: string]: {
+      [asset: string]: {
+        amount: string;
+        unit: string;
+      };
+    };
+  };
+};
+
+/**
+ * Default state of the {@link MultichainBalancesController}.
+ */
+export const defaultState: MultichainBalancesControllerState = { balances: {} };
```

Comment:
Nit: As per our controller example in guidelines https://github.com/MetaMask/core/blob/main/examples/example-controllers/src/pet-names-controller.ts 
```suggestion
/**
 * Constructs the default {@link MultichainBalancesController} state. This allows
 * consumers to provide a partial state object when initializing the controller
 * and also helps in constructing complete state objects for this controller in
 * tests.
 *
 * @returns The default {@link MultichainBalancesController} state.
 */
export function getDefaultMultichainBalancesControllerState(): MultichainBalancesControllerState {
  return { balances: {} }
}
```

Reply:
done in [bcaee6c](https://github.com/MetaMask/core/pull/4965/commits/bcaee6c42b4c196809fe2c59cd611ca155794b28)